components/Extensions/tests/DispatcherQueueTimerExtensionTests.cs

```diff
@@ -54,20 +148,72 @@ public async Task DispatcherQueueTimer_Debounce_Interrupt()
             TimeSpan.FromMilliseconds(60));
 
         Assert.AreEqual(true, debounceTimer.IsRunning, "Expected time to be running.");
+        Assert.AreEqual(0, triggeredCount, "Function shouldn't have run yet.");
+        Assert.AreEqual(0, customTriggeredCount, "Custom Function shouldn't have run yet.");
+        Assert.IsNull(triggeredValue, "Function shouldn't have run yet.");
 
         await Task.Delay(TimeSpan.FromMilliseconds(110));
 
         Assert.AreEqual(false, debounceTimer.IsRunning, "Expected to stop the timer.");
         Assert.AreEqual(value2, triggeredValue, "Expected to execute the last action.");
         Assert.AreEqual(1, triggeredCount, "Expected to postpone execution.");
+        Assert.AreEqual(1, customTriggeredCount, "Expected to postpone execution of custom event handler.");
+    }
+
+    [TestCategory("DispatcherQueueTimerExtensions")]
+    [UIThreadTestMethod]
+    public async Task DispatcherQueueTimer_Debounce_Immediate()
+    {
+        var debounceTimer = DispatcherQueue.GetForCurrentThread().CreateTimer();
+
+        // Test custom event handler too
+        var customTriggeredCount = 0;
+        debounceTimer.Tick += (s, o) =>
+        {
+            customTriggeredCount++;
+        };
+
+        var triggeredCount = 0;
+        string? triggeredValue = null;
+
+        var value = "He";
+        debounceTimer.Debounce(
+            () =>
+            {
+                triggeredCount++;
+                triggeredValue = value;
+            },
+            TimeSpan.FromMilliseconds(60), true);
+
+        Assert.AreEqual(true, debounceTimer.IsRunning, "Expected time to be running.");
+        Assert.AreEqual(1, triggeredCount, "Function should have run right away.");
+        Assert.AreEqual(0, customTriggeredCount, "Custom Function won't have run as cooldown hasn't elapsed.");
+        Assert.AreEqual(value, triggeredValue, "Should have expected immediate set of value");
+
+        await Task.Delay(TimeSpan.FromMilliseconds(80));
+
+        Assert.AreEqual(false, debounceTimer.IsRunning, "Expected to stop the timer.");
+        Assert.AreEqual(1, customTriggeredCount, "Custom Function should have run now that cooldown expired.");
     }
 
+    /// <summary>
+    /// Tests the immediate mode of the Debounce function ignoring subsequent inputs that come after the first within the specified time window.
```

Comment:
Just adding a paragraph marker so it'll render properly in Visual Studio. 
```suggestion
    /// Tests the immediate mode of the Debounce function ignoring subsequent inputs that come after the first within the specified time window.
    /// <para />
```